sw/device/silicon_creator/rom/rom.c

```diff
@@ -156,9 +157,14 @@ static rom_error_t rom_init(void) {
   // Update epmp config for debug rom according to lifecycle state.
   rom_epmp_config_debug_rom(lc_state);
 
-  // Re-initialize the watchdog timer.
-  watchdog_init(lc_state);
-  SEC_MMIO_WRITE_INCREMENT(kWatchdogSecMmioInit);
+  uint32_t reset_reasons = rstmgr_reason_get();
+  if (reset_reasons & ~(1U << RSTMGR_RESET_INFO_LOW_POWER_EXIT_BIT)) {
```

Comment:
Should we also check the lock bit status?

Reply:
That raises an interesting question at a higher layer: Should the lock bits actually survive deep sleep? Shouldn't ROM be able to configure the IPs it needs?

Reply:
I do not think that we should base the decision on whether owner code has locked the watchdog configuration registers, I think it makes sense that ROM configures watchdog (non-locked) on boot, and then refrains from touching it when coming out of deep sleep later (as the watchdog IP is not being reset, and does not need re-configuring.)

To answer Alex.  I find it reassuring that e.g. watchdog, sysrst keyboard combinations, pinmux and other things, once configured by Ti50 on first cold boot, cannot be accidentally reconfigured by a bug in Ti50 even after going in and out of deep sleep many times.  I agree that it is weird if owner code reconfigures pinmux such that e.g. serial console cannot work, and locks pinmux that way that, but I am OK if it would take an explicit reset to get out of such a situation.

Reply:
Personally, I lean the other way, as ROM is generally written for only a limited set of known states, constructing a narrow gate to pass through and jump to the next stage. More branching paths creates a larger surface for coverage closure and increases the risk of bugs to something that can't be updated. Meanwhile, the applications likely would have fewer risks of deep-sleep-specific init bugs if they simply had to redo init, instead of having two paths, but that wouldn't be as fast, hehe.

Ultimately, I guess either option gives us ways to shoot ourselves in the foot. They both still put it on the application developer to be compatible with whatever ROM does. Perhaps the most important thing is to precisely document that interface.